test/test_detect_breaks.py

```diff
@@ -0,0 +1,141 @@
+    time_stamps = list(range(-5, 5, 2))
+    stream = MockStreamData(time_stamps=time_stamps, tdiff=1)
+    # if diff > 1 -> 4
+    breaks = _detect_breaks(stream, threshold_seconds=0, threshold_samples=1)
+    assert breaks.size == len(time_stamps) - 1
+
+
+def test_detect_breaks_gap_in_negative():
+    time_stamps = [-4, 1, 2, 3, 4]
+    stream = MockStreamData(time_stamps=time_stamps, tdiff=1)
+    # if diff > 1 -> 1
+    b_breaks = _detect_breaks(stream, threshold_seconds=1, threshold_samples=0)
+    breaks = np.where(b_breaks)[0]
+    assert breaks.size == 1
+    assert breaks[0] == 0
+    time_stamps = [-4, -2, -1, 0, 1, 2, 3, 4]
+    stream = MockStreamData(time_stamps=time_stamps, tdiff=1)
+    # if diff > 1 -> 1
+    b_breaks = _detect_breaks(stream, threshold_seconds=1, threshold_samples=0)
+    breaks = np.where(b_breaks)[0]
+    assert breaks.size == 1
+    assert breaks[0] == 0
+    # if diff > 0.1 -> 7
+    b_breaks = _detect_breaks(stream, threshold_seconds=0.1, threshold_samples=0)
+    breaks = np.where(b_breaks)[0]
+    assert breaks.size == len(time_stamps) - 1
+
+
+def test_detect_breaks_gap_in_positive():
+    time_stamps = [1, 3, 4, 5, 6]
+    stream = MockStreamData(time_stamps=time_stamps, tdiff=1)
+    # if diff > 1 -> 1
+    b_breaks = _detect_breaks(stream, threshold_seconds=1, threshold_samples=0)
+    breaks = np.where(b_breaks)[0]
+    assert breaks.size == 1
+    assert breaks[0] == 0
+    # if diff > 0.1 -> 4
+    b_breaks = _detect_breaks(stream, threshold_seconds=0.1, threshold_samples=0)
+    breaks = np.where(b_breaks)[0]
+    assert breaks.size == len(time_stamps) - 1
+
+
+# Non-monotonic timeseries data: segment at negative time-intervals or positive
+# time-intervals exceeding threshold
+
+
+def test_detect_breaks_reverse():
+    time_stamps = list(reversed(range(-5, 5)))
+    stream = MockStreamData(time_stamps=time_stamps, tdiff=1)
+    # if diff < 0 -> 9
+    b_breaks = _detect_breaks(stream, threshold_seconds=1, threshold_samples=0)
+    breaks = np.where(b_breaks)[0]
+    assert breaks.size == len(time_stamps) - 1
+
+
+def test_detect_breaks_non_monotonic_gap_in_negative():
+    time_stamps = [-4, -7, -6, -5, -3, -2, -1, 0]
+    stream = MockStreamData(time_stamps=time_stamps, tdiff=1)
+    # if diff < 0 or diff > 1 -> 2
+    b_breaks = _detect_breaks(stream, threshold_seconds=1, threshold_samples=0)
+    breaks = np.where(b_breaks)[0]
+    assert breaks.size == 2
+    assert breaks[0] == 0
+    assert breaks[1] == 3
+    # if diff < 0 or diff > 2 -> 1
+    b_breaks = _detect_breaks(stream, threshold_seconds=2, threshold_samples=0)
+    breaks = np.where(b_breaks)[0]
+    assert breaks.size == 1
+    assert breaks[0] == 0
+    # if diff < 0 or diff > 0.1 -> 7
+    b_breaks = _detect_breaks(stream, threshold_seconds=0.1, threshold_samples=0)
+    breaks = np.where(b_breaks)[0]
+    assert breaks.size == len(time_stamps) - 1
+
+
+def test_detect_breaks_non_monotonic_gap_in_positive():
```

Comment:
Same comment on duplication here, this test is identical to the previous one except for `time_stamps`. Do you think it is more readable like this, or would parameterizing be an option? I'm not insisting on parameterization, I'm just curious what you think.